src/docs/development/platform-integration/platform-channels.md

```diff
@@ -646,3 +661,66 @@ classes, or create your own codec.
 [BinaryCodec]: {{site.api}}/flutter/services/BinaryCodec-class.html
 [StringCodec]: {{site.api}}/flutter/services/StringCodec-class.html
 [JSONMessageCodec]: {{site.api}}/flutter/services/JSONMessageCodec-class.html
+
+## Channels and Platform Threading
+
+When writing code on the platform side, which is the Android/iOS side using
+Java/Kotlin/Objective-C/Swift, all channel methods must be invoked on the
+platform's main thread. Please see official documentation about [the main
+thread on Android] and [the main thread on iOS].
+
+[the main thread on Android]: https://developer.android.com/guide/components/processes-and-threads#Threads
```

Comment:
On Android "main thread" is an overloaded term. The documentation linked here mentions that colloquially the "main thread" often means the "UI Thread" and then proceeds to talk about the UI Thread, but points out that "main thread" is actually a distinct term and links to another page about the annotations where `@MainThread` is a separate thing.

I'm pretty sure on Android we really mean `@UiThread` whenever we refer to "main thread," and not `@MainThread`, so I think it's worth pointing out the discrepancy. This is mostly what I was thinking of when I was talking about term overloading earlier.

Reply:
I gave it another pass. Let me know what you think.

Reply:
LG!